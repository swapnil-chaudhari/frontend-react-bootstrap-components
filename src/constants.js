export const FILTER_PERIODS = {
    YEAR: 365,
    MONTH: 27,
    WEEK: 6,
    DAYS_90: 90,
};

export const SEARCH_TYPES = {
    ARTISTS: 'artists',
    TRACKS: 'tracks'
};

export const KEYS = {
    ENTER: 13,
};
