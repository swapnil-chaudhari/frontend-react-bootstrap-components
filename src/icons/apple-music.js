import React from 'react';
import { ICON_IDS } from 'src/svg-definitions/svgs';

const AppleMusic = () => (
    /* eslint-disable max-len */
    <svg id={ ICON_IDS.APPLE_MUSIC } viewBox="0 0 24 24" className="Icons-apple-music">
        <defs>
            <linearGradient x1="41.5902151%" y1="6.80428608%" x2="75.9989248%" y2="77.1777806%" id="apple-music-linear-gradient">
                <stop stopColor="#FF505C" offset="0%" />
                <stop stopColor="#38ADFB" offset="100%" />
            </linearGradient>
        </defs>
        <g>
            <path d="M6,0 L18,0 C21.3137085,-6.08718376e-16 24,2.6862915 24,6 L24,18 C24,21.3137085 21.3137085,24 18,24 L6,24 C2.6862915,24 4.05812251e-16,21.3137085 0,18 L0,6 C-4.05812251e-16,2.6862915 2.6862915,6.08718376e-16 6,0 Z" fill="#D9D2D0" />
            <path d="M6,1 L18,1 C20.7614237,1 23,3.23857625 23,6 L23,18 C23,20.7614237 20.7614237,23 18,23 L6,23 C3.23857625,23 1,20.7614237 1,18 L1,6 C1,3.23857625 3.23857625,1 6,1 Z" fill="#FFFFFF" />
            <path d="M17.1082436,15.5069523 C17.1082436,15.8893303 17.0567396,16.2654654 16.8866206,16.612727 C16.6212973,17.1542991 16.1881961,17.4945375 15.6138493,17.6576334 C15.294681,17.7489359 14.9677091,17.8012202 14.633714,17.8160471 C13.7612686,17.8574064 13.0074384,17.265891 12.8513659,16.4082719 C12.7218257,15.7004824 13.0612834,14.9201192 13.8049689,14.5556896 C14.1015067,14.4089813 14.4183339,14.3270431 14.7375022,14.2583712 C15.0839831,14.1842367 15.4328052,14.1179058 15.7769451,14.0383087 C16.0282218,13.9805619 16.1959998,13.8276107 16.245943,13.5654086 C16.256868,13.5076617 16.2615502,13.4475738 16.2615502,13.3882662 C16.2623306,11.7245318 16.2623306,10.0615778 16.2607699,8.39862372 C16.2607699,8.34165721 16.2514055,8.28391033 16.2381393,8.2292849 C16.2014623,8.09116061 16.1007955,8.00688138 15.9587695,8.01468502 C15.8128418,8.02326901 15.6676943,8.04667991 15.5241076,8.07555335 C14.8264635,8.21289728 14.1311605,8.35414302 13.4342967,8.4946084 C12.7241668,8.63741487 12.0124762,8.78178207 11.3015659,8.9253689 C10.8817309,9.01042849 10.4611154,9.09470772 10.0412804,9.17976731 C10.0264535,9.18288876 10.0116266,9.18757094 9.9975801,9.19147276 C9.74396227,9.26170545 9.65187949,9.37797957 9.64017405,9.64018161 C9.63861333,9.67998014 9.63939369,9.71977866 9.63939369,9.76035755 C9.63861333,12.1451476 9.63939369,14.5307179 9.63783297,16.9155079 C9.63783297,17.300227 9.59569339,17.6810443 9.44040124,18.039231 C9.18522269,18.6276249 8.73495351,18.9936152 8.12549036,19.1707577 C7.80398099,19.2636209 7.47388764,19.3166856 7.14067283,19.328391 C6.26198461,19.3611663 5.53234564,18.7774546 5.3809553,17.9128121 C5.25063476,17.1675653 5.59165319,16.367693 6.43756619,16.0095062 C6.76531845,15.8721623 7.10867797,15.7964671 7.45281785,15.7246737 C7.71580003,15.6692679 7.97956257,15.6193246 8.24176438,15.5631385 C8.59214716,15.4874432 8.77553236,15.2666005 8.79035925,14.9084137 C8.79191997,14.8615919 8.79113961,14.8171112 8.79113961,14.7710698 C8.79113961,12.0569665 8.79113961,9.34286322 8.79270033,6.6295403 C8.79270033,6.51638763 8.80440577,6.40011351 8.83171846,6.29008229 C8.89570819,6.02866061 9.08143447,5.87961124 9.33115049,5.81484109 C9.56525925,5.75397276 9.80326982,5.71183314 10.0405,5.66345062 C10.7123922,5.52766742 11.3850647,5.39266459 12.0561765,5.25766175 C12.7506991,5.11719637 13.4452218,4.97751135 14.1389641,4.83704597 C14.7523291,4.71296822 15.3664744,4.58732974 15.9814001,4.46637344 C16.1827336,4.42657491 16.3864082,4.38599603 16.5893025,4.3696084 C16.8733544,4.34697787 17.0684451,4.52568104 17.0973185,4.81129398 C17.1043418,4.87918558 17.1082436,4.94785755 17.1082436,5.01652951 C17.1098043,6.76844495 17.1098043,8.51958002 17.1090239,10.2707151 C17.1090239,12.0163876 17.1098043,13.7628405 17.1082436,15.5069523 Z" fill="url(#apple-music-linear-gradient)" />
        </g>
    </svg>
    /* eslint-enable max-len */
);

export default AppleMusic;
