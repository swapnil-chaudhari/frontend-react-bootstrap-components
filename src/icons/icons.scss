@import "src/styles/variables";

.Icons {
    &-attention {
        fill: $red-200;
        height: 16px;
        width: 16px;
    }

    &-audio-player-mute {
        fill: $white;
        height: 21px;
        width: 26px;
    }

    &-audio-player-next {
        height: 30px;
        width: 30px;
    }

    &-audio-player-next-lite {
        height: 12px;
        width: 12px;
    }

    &-audio-player-pause {
        height: 30px;
        width: 30px;
    }

    &-audio-player-pause-lite {
        height: 14px;
        width: 12px;
    }

    &-audio-player-play {
        height: 30px;
        width: 30px;
    }

    &-audio-player-previous {
        height: 30px;
        width: 30px;
    }

    &-audio-player-previous-lite {
        height: 12px;
        width: 12px;
    }

    &-audio-player-volume {
        fill: $white;
        height: 20px;
        width: 23px;
    }

    &-calendar {
        height: 15px;
        width: 15px;
    }

    &-complete-checkmark {
        fill: $turtle;
        height: 16px;
        width: 16px;
    }

    &-chevron {
        fill: $grey_800;
        height: 16px;
        vertical-align: middle;
        width: 16px;
    }

    &-clean-version {
        height: 15px;
        margin-left: 10px;
        position: relative;
        top: 3px;
        width: 42px;
    }

    &-close {
        cursor: pointer;
        fill: $grey_200;
        height: 16px;
        width: 16px;
    }

    &-digital,
    &-physical,
    &-film {
        height: 57px;
        width: 55px;
    }

    &-explicit {
        height: 15px;
        margin-left: 10px;
        position: relative;
        top: 3px;
        width: 54px;
    }

    &-incomplete-alert {
        fill: $fox;
        height: 16px;
        width: 16px;
    }

    &-info {
        height: 12px;
        margin-left: 6px;
        margin-right: 6px;
        width: 12px;
    }

    &-play {
        fill: $rhino;
        height: 12px;
        width: 9px;
    }

    &-status-red,
    &-status-complete,
    &-status-in-progress,
    &-status-submitted {
        height: 12px;
        width: 12px;
    }

    &-stop {
        fill: $bluejay;
        height: 8px;
        width: 8px;
    }

    &-upload {
        height: 16px;
        margin-right: 5px;
        vertical-align: text-top;
        width: 16px;
    }

    &-right-direction-arrow {
        height: 22px;
        width: 12px;
    }

    &-loading-ring {
        height: 120px;
        width: 120px;
    }

    &-trash-can {
        fill: $grey_700;
        height: 16px;
        width: 16px;
    }

    &-close-round {
        height: 25px;
        width: 25px;
    }

    &-search {
        fill: $gorilla;
        height: 18px;
        width: 16px;
    }

    &-three-right-directed-arrows {
        height: 16px;
        width: 30px;
    }

    &-left-pagination-arrow {
        fill: $grey_900;
        height: 13px;
        width: 8px;
    }

    &-right-pagination-arrow {
        fill: $grey_900;
        height: 13px;
        width: 8px;
    }

    &-logo {
        height: 32px;
        width: 23px;
    }

    &-loading-arrows {
        animation: roll 1.25s ease infinite;
        fill: $timberwolf;
        height: 20px;
        width: 15px;
    }

    &-edit {
        fill: $grey_700;
        height: 16px;
        width: 16px;
    }

    &-external-link {
        fill: $rhino;
        height: 16px;
        width: 16px;
    }

    &-up-and-down-arrows {
        fill: $grey_400;
        height: 20px;
        width: 8px;
    }

    &-download-box {
        height: 120px;
        width: 120px;
    }

    &-triple-checkmark {
        height: 16px;
        width: 16px;
    }

    &-up-and-down-carets {
        fill: $rhino;
        height: 16px;
        width: 16px;
    }

    &-tracks-complete-checkmark {
        fill: $koala;
    }

    &-view-more-next-arrow {
        fill: $bluejay;
        height: 16px;
        width: 16px;
    }

    &-plus {
        fill: $rhino;
        height: 16px;
        width: 16px;
    }

    &-man {
        fill: $grey_700;
        height: 16px;
        width: 15px;
    }

    &-woman {
        fill: $grey_700;
        height: 18px;
        width: 12px;
    }

    &-down-direction-arrow {
        fill: $rhino;
        height: 16px;
        width: 16px;
    }

    &-up-direction-arrow {
        fill: $rhino;
        height: 16px;
        width: 16px;
    }
    &-three-dots {
        fill: $rhino;
        height: 30px;
        width: 30px;
    }

    &-download-symbol {
        fill: $rhino;
        height: 16px;
        width: 16px;
    }

    &-refresh {
        fill: $rhino;
        height: 16px;
        width: 16px;
    }

    &-server-error {
        height: 50px;
        width: 80px;
    }

    &-new-product {
        height: 67px;
        width: 68px;
    }

    &-advertise {
        height: 65px;
        width: 68px;
    }

    &-go-to-catalog {
        height: 54px;
        width: 66px;
    }

    &-transparent-logo {
        height: 60px;
        width: 60px;
    }

    &-history {
        fill: $timberwolf;
        height: 16px;
        width: 15px;
    }

    &-pdf {
        height: 28px;
        width: 23px;
    }

    &-xls {
        height: 28px;
        width: 23px;
    }

    &-royalty-share-logo {
        height: 40px;
        width: 354px;
    }

    &-error-404 {
        height: 288px;
        width: 486px;
    }

    &-maintenance {
        height: 131px;
        width: 220px;
    }

    &-logo-and-name {
        height: 40px;
        width: 110px;
    }

    &-apple-music {
        height: 16px;
        width: 16px;
    }

    &-spotify {
        height: 16px;
        width: 16px;
    }

    &-facebook {
        height: 16px;
        width: 16px;
    }

    &-aan {
        height: 12px;
        width: 32px;
    }

    &-twitter {
        height: 16px;
        width: 14px;
    }

    &-instagram {
        height: 16px;
        width: 16px;
    }

    &-youtube {
        height: 12px;
        width: 16px;
    }

    &-camera {
        height: 16px;
        width: 16px;
    }

    &-playlist-placeholder {
        height: 135px;
        width: 135px;
    }

    @keyframes roll {
        0% {
            transform: rotate(0deg);
        }

        100% {
            transform: rotate(360deg);
        }
    }
}

/* stylelint-disable selector-no-id */
#audio-player-play-circle,
#audio-player-previous-circle,
#audio-player-previous-lite-path,
#audio-player-next-circle,
#audio-player-next-lite-path {
    fill: $rhino;
}

#audio-player-play-triangle,
#audio-player-previous-path,
#audio-player-next-path {
    fill: $white;
}

#audio-player-pause-lite-path {
    fill: #ff893e;
}
/* stylelint-enable selector-no-id */
