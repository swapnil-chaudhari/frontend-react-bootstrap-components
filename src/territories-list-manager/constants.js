export const TERRITORIES_KEYS = {
    ID: 'orchId',
    NAME: 'territoryName',
    CONTINENT: 'continent'
};
