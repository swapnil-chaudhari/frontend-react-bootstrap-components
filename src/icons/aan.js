import React from 'react';
import { Icon } from 'src/index';
import { ICON_IDS } from 'src/svg-definitions/svgs';

const AAN = () => <Icon className="Icons-aan" id={ ICON_IDS.AAN } />;

export default AAN;
