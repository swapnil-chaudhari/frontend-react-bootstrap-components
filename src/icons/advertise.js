import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'src/index';
import { ICON_IDS } from 'src/svg-definitions/svgs';
import cx from 'classnames';

const Advertise = ({ className }) =>
    <Icon
        className={ cx('Icons-advertise', className) }
        id={ ICON_IDS.ADVERTISE }
    />;

Advertise.propTypes = {
    className: PropTypes.string
};

export default Advertise;
