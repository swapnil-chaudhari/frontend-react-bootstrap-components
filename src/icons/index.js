import './icons.scss';

export { default as Aan } from './aan';
export { default as Advertise } from './advertise';
export { default as AppleMusic } from './apple-music';
export { default as ArtworkEmpty } from './artwork-empty';
export { default as Attention } from './attention';
export { default as AudioDisabled } from './audio-disabled';
export { default as AudioEnabled } from './audio-enabled';
export { default as AudioPlayerMute } from './audio-player-mute';
export { default as AudioPlayerNext } from './audio-player-next';
export { default as AudioPlayerNextLite } from './audio-player-next-lite';
export { default as AudioPlayerPause } from './audio-player-pause';
export { default as AudioPlayerPauseLite } from './audio-player-pause-lite';
export { default as AudioPlayerPlay } from './audio-player-play';
export { default as AudioPlayerPrevious } from './audio-player-previous';
export { default as AudioPlayerPreviousLite } from './audio-player-previous-lite';
export { default as AudioPlayerVolume } from './audio-player-volume';
export { default as Calendar } from './calendar';
export { default as Camera } from './camera';
export { default as Chevron } from './chevron';
export { default as CleanVersion } from './clean-version';
export { default as Close } from './close';
export { default as CloseRound } from './close-round';
export { default as CompleteCheckmark } from './complete-checkmark';
export { default as Digital } from './digital';
export { default as DigitalAudio } from './digital-audio';
export { default as DownDirectionArrow } from './down-direction-arrow';
export { default as DownloadBox } from './download-box';
export { default as DownloadSymbol } from './download-symbol';
export { default as Edit } from './edit';
export { default as Error404 } from './error-404.js';
export { default as Explicit } from './explicit';
export { default as ExternalLink } from './external-link';
export { default as Facebook } from './facebook';
export { default as FilledChevronDown } from './filled-chevron-down';
export { default as FilledChevronUp } from './filled-chevron-up';
export { default as Film } from './film';
export { default as Gear } from './gear.js';
export { default as GoToCatalog } from './go-to-catalog';
export { default as History } from './history';
export { default as IncompleteAlert } from './incomplete-alert';
export { default as ImportArrow } from './import-arrow';
export { default as Info } from './info';
export { default as Instagram } from './instagram';
export { default as LeftPaginationArrow } from './left-pagination-arrow';
export { default as LoadingArrows } from './loading-arrows';
export { default as LoadingRing } from './loading-ring';
export { default as Logo } from './logo';
export { default as LogoAndName } from './logo-and-name.js';
export { default as Maintenance } from './maintenance.js';
export { default as Man } from './man';
export { default as NewProduct } from './new-product';
export { default as Pdf } from './pdf';
export { default as Physical } from './physical';
export { default as PhysicalAudio } from './physical-audio';
export { default as Play } from './play';
export { default as PlaylistPlaceholder } from './playlist-placeholder';
export { default as Plus } from './plus';
export { default as Refresh } from './refresh';
export { default as RightDirectionArrow } from './right-direction-arrow';
export { default as RightPaginationArrow } from './right-pagination-arrow';
export { default as RoyaltyShareLogo } from './royalty-share-logo';
export { default as Search } from './search';
export { default as ServerError } from './server-error';
export { default as Spotify } from './spotify';
export { default as StatusComplete } from './status-complete';
export { default as StatusInProgress } from './status-in-progress';
export { default as StatusRed } from './status-red';
export { default as StatusSubmitted } from './status-submitted';
export { default as Stop } from './stop';
export { default as SuccessCheckmark } from './success-checkmark';
export { default as ThreeDots } from './three-dots';
export { default as ThreeRightDirectedArrows } from './three-right-directed-arrows';
export { default as TracksCompleteCheckmark } from './tracks-complete-checkmark';
export { default as TracksEmpty } from './tracks-empty';
export { default as TransparentLogo } from './transparent-logo';
export { default as TrashCan } from './trash-can';
export { default as TripleCheckmark } from './triple-checkmark';
export { default as Twitter } from './twitter';
export { default as UpAndDownArrows } from './up-and-down-arrows';
export { default as UpAndDownCarets } from './up-and-down-carets';
export { default as UpDirectionArrow } from './up-direction-arrow';
export { default as Upload } from './upload';
export { default as UploadArrow } from './upload-arrow';
export { default as VideoError } from './video-error';
export { default as Video } from './video';
export { default as ViewMoreNextArrow } from './view-more-next-arrow';
export { default as Woman } from './woman';
export { default as Xls } from './xls';
export { default as YouTube } from './youtube';
