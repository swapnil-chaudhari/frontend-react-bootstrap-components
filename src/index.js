import * as Icons from './icons';
import {
    Tooltip,
    Overlay,
    OverlayTrigger,
    Nav,
    NavItem,
    MenuItem,
    NavDropdown,
    Popover
} from 'react-bootstrap';

// Analytics components
export { default as Snapshot } from './snapshot/snapshot';
export { default as SnapshotItem } from './snapshot/snapshot-item';

// Bootstrap and bootstrap-based components
export { default as AlertMessage } from './alert-message/alert-message';
export { default as Button } from './button/button';
export { default as LoadingButton } from './loading-button/loading-button';
export { default as ButtonsRow } from './buttons-row/buttons-row';
export { default as CollapsableAlert } from './collapsable-alert/collapsable-alert';
export { default as DismissibleAlert } from './dismissible-alert/dismissible-alert';
export { default as DropdownButton } from './dropdown-button/dropdown-button';
export {
    MenuItem,
    Nav,
    NavDropdown,
    NavItem,
    Overlay,
    OverlayTrigger,
    Popover,
    Tooltip
};

// Chart components
export { default as ChartContainer } from './chart/chart-container';
export { default as ChartRectClipPath } from './chart/chart-rect-clip-path';
export { default as ChartOverlayTrigger } from './chart/chart-overlay-trigger';
export { default as ChartXAxis } from './chart/chart-x-axis.js';
export { default as ChartYAxis } from './chart/chart-y-axis';
export { default as ChartGrid } from './chart/chart-grid';
export { default as ChartLine } from './chart/chart-line';
export { default as ChartGraph } from './chart/chart-graph';
export { default as ChartGraphContainer } from './chart/chart-graph-container';

export { default as DoughnutChart } from './doughnut-chart/doughnut-chart';
export { default as GroupedBarChart } from './grouped-bar-chart/grouped-bar-chart';

// Form field components
export {
    default as CharacterLimitIndicator
} from './character-limit-indicator/character-limit-indicator';
export { default as Checkbox } from './checkbox/checkbox';
export { default as DayInput } from './day-input/day-input';
export { default as SingleDateInput } from './single-date-input/single-date-input';
export { default as RangeDateInput } from './range-date-input/range-date-input';
export { default as FormField } from './form-field/form-field';
export { default as RadioButton } from './radio-button-group/radio-button/radio-button';
export { default as RadioButtonGroup } from './radio-button-group/radio-button-group';
export { default as SelectInput } from './select-input/select-input';
export {
    default as SelectInputWithAddNew
} from './select-input-with-add-new/select-input-with-add-new';
export { default as TagsInput } from './tags-input/tags-input';
export { default as FullWidthButton } from './full-width-button/full-width-button';

// Iconography
export { default as Help } from './help/help';
export { default as Icon } from './icon/icon';
export { Icons };
export { default as SVGDefinitions } from './svg-definitions/svg-definitions';

// Loading states
export { default as LoadingIndicator } from './loading-indicator/loading-indicator';
export { default as LoadingStateWrapper } from './loading-state-wrapper/loading-state-wrapper';

// Navigation items
export { default as MainNavigation } from './main-navigation/main-navigation';
export { default as Pagination } from './pagination/pagination';
export { default as PaginationItem } from './pagination/pagination-item';
export { default as SubNavigation } from './sub-navigation/sub-navigation';
export { default as SubNavigationItem } from './sub-navigation/sub-navigation-item';

// Track/Artist search
export { default as SearchResults } from './search-row/search-results/search-results';
export { default as SearchRow } from './search-row/search-row';

// Tables
export { default as ClickableTable } from './clickable-table/clickable-table';

// Tile
export { default as Tile } from './tile/tile';

// Territories
export {
    default as TerritoriesListManager
} from './territories-list-manager/territories-list-manager';
export {
    default as ListHeaderSearch
} from './territories-list-manager/list-header-search/list-header-search';

export { default as MaintenanceScreen } from './maintenance-screen/maintenance-screen.js';
export { default as Modal } from './modal/modal.js';
export { default as ToggleInput } from './toggle-input/toggle-input.js';
