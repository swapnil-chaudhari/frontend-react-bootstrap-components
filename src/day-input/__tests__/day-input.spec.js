import React from 'react';
import { Calendar as CalendarIcon, RightPaginationArrow, LeftPaginationArrow } from 'src/icons';
import { shallow, mount } from 'enzyme';
import DayPickerInput from 'react-day-picker/DayPickerInput';
import { KEYS } from 'src/constants';
import DayInput from '../day-input';
import Input from '../input/input';

describe('<DayInput>', () => {
    const commonProps = {
        name: 'my-day-input',
        formatMessage: m => m.defaultMessage,
        messages: {
            todayLabel: { defaultMessage: 'todayLabel' }
        }
    };

    describe('when rendered with required props', () => {
        const props = { ...commonProps, disabled: false };
        let component;

        beforeAll(() => {
            component = shallow(<DayInput { ...props } />);
        });

        test('renders <DayInput> with the right props', () => {
            expect(component.props()).toMatchSnapshot();
        });

        test('renders the <CalendarIcon> inside <Input>', () => {
            const navBar = component.find(DayPickerInput).dive().find(Input).dive();
            expect(navBar.find(CalendarIcon)).toHaveLength(1);
        });
    });

    describe('when today button is hidden', () => {
        const props = {
            ...commonProps,
            renderTodayButton: false,
            dayPickerInputOptions: {
                showOverlay: true
            }
        };
        let component;

        beforeAll(() => {
            component = mount(<DayInput { ...props } />);
        });

        test('does not render the Today button', () => {
            expect(component.find('.DayPicker-wrapper')).toHaveLength(1);
            expect(component.find('.DayPicker-TodayButton')).toHaveLength(0);
        });
    });

    describe('when rendered <Navbar>', () => {
        const props = {
            ...commonProps,
            disabled: false,
            dayPickerInputOptions: {
                showOverlay: true
            }
        };
        let component;

        beforeAll(() => {
            component = mount(<DayInput { ...props } />);
        });

        test('renders the <LeftPaginationArrow> and <RightPaginationArrow>', () => {
            expect(component.find(LeftPaginationArrow)).toHaveLength(1);
            expect(component.find(RightPaginationArrow)).toHaveLength(1);
        });

        test('renders Today button', () => {
            expect(component.find('.DayPicker-TodayButton')).toHaveLength(1);
        });
    });

    describe('when date is changed', () => {
        let instance;

        beforeEach(() => {
            const props = { ...commonProps, disabled: false, onChange: jest.fn() };
            instance = shallow(<DayInput { ...props } />).instance();
        });

        test('has correct parameters if correct date is passed to onChange handler', () => {
            instance.handleChange(new Date(2000, 0, 1));
            expect(instance.props.onChange.mock.calls[0][0]).toEqual('2000-01-01');
            expect(instance.props.onChange.mock.calls[0][1].dateMoment.isValid()).toBeTruthy();
        });

        test('has correct parameters if incorrect date is passed to onChange handler', () => {
            instance.handleChange(null);
            expect(instance.props.onChange.mock.calls[0][0]).toEqual('');
            expect(instance.props.onChange.mock.calls[0][1].dateMoment.isValid()).toBeFalsy();
        });

        test('has correct parameters if undefined date is passed to onChange handler', () => {
            instance.handleChange(undefined);
            expect(instance.props.onChange.mock.calls[0][0]).toEqual('');
            expect(instance.props.onChange.mock.calls[0][1].dateMoment.isValid()).toBeFalsy();
        });
    });

    describe('when event allowed in onChange handler', () => {
        describe('when the date is invalid', () => {
            let instance;

            beforeEach(() => {
                const props = { ...commonProps, onChange: jest.fn(), returnEventIfError: true };
                instance = shallow(<DayInput { ...props } />).instance();
            });

            test('has event as a first argument', () => {
                instance.handleChange(undefined);
                expect(instance.props.onChange.mock.calls[0][0]).toEqual({ target: { value: '' } });
                expect(instance.props.onChange.mock.calls[0][1].dateMoment.isValid()).toBeFalsy();
            });
        });

        describe('when date is correct', () => {
            let instance;

            beforeEach(() => {
                const props = { ...commonProps, onChange: jest.fn(), returnEventIfError: true };
                instance = shallow(<DayInput { ...props } />).instance();
            });

            test('has string date as a first argument', () => {
                instance.handleChange(new Date(2000, 0, 1));
                expect(instance.props.onChange.mock.calls[0][0]).toEqual('2000-01-01');
                expect(instance.props.onChange.mock.calls[0][1].dateMoment.isValid()).toBeTruthy();
            });
        });
    });

    describe('with dateFormat', () => {
        test('has default format if nothing passed in props', () => {
            const instance = shallow(<DayInput { ...commonProps } />).instance();
            expect(instance.dateFormat).toEqual('YYYY-MM-DD');
        });

        test('has format from props', () => {
            const component = shallow(<DayInput { ...commonProps } dateFormat="DD-MM-YYYY" />);
            expect(component.instance().dateFormat).toEqual('DD-MM-YYYY');
        });
    });

    describe('when date is parsed', () => {
        const props = { ...commonProps, disabled: false };
        let instance;

        beforeAll(() => {
            instance = shallow(<DayInput { ...props } />).instance();
        });

        test('returns date if correct string is passed to parseDate handler', () => {
            expect(instance.parseDate('2000-01-05')).toEqual(new Date(2000, 0, 5));
        });

        test('returns undefined if incomplete date string is passed to parseDate handler', () => {
            expect(instance.parseDate('2000-01')).toBeUndefined();
        });

        test('returns undefined if incorrect string is passed to parseDate handler', () => {
            expect(instance.parseDate(null)).toBeUndefined();
            expect(instance.parseDate(undefined)).toBeUndefined();
            expect(instance.parseDate('Invalid date string')).toBeUndefined();
        });
    });

    describe('when date is formatted', () => {
        const props = { ...commonProps, disabled: false };
        let instance;

        beforeAll(() => {
            instance = shallow(<DayInput { ...props } />).instance();
        });

        test('returns date string if correct date is passed to formatDate handler', () => {
            expect(instance.formatDate(new Date(2000, 0, 5))).toEqual('2000-01-05');
        });

        test('returns empty string if incorrect date is passed to formatDate handler', () => {
            expect(instance.formatDate(null)).toEqual('');
        });
    });

    describe('when focusing the input field', () => {
        const props = { ...commonProps, disabled: false, focus: true };
        let instance;

        beforeAll(() => {
            instance = mount(<DayInput { ...props } />).instance();
            jest.spyOn(instance.dayPickerInput.input, 'focus');
            instance.componentDidMount();
        });

        afterAll(() => {
            jest.restoreAllMocks();
        });

        test('calls focus method', () => {
            expect(instance.dayPickerInput.input.focus).toHaveBeenCalled();
        });
    });

    describe('when handleYearMonthChange is called', () => {
        const mayMonth = new Date(2000, 4).getMonth();
        const props = { ...commonProps, disabled: false };
        let instance;

        beforeAll(() => {
            instance = mount(<DayInput { ...props } />).instance();
            instance.handleYearMonthChange(mayMonth);
        });

        test('has May month with value 4 in state', () => {
            expect(instance.state.date).toEqual(mayMonth);
        });
    });

    describe('when getLocaleUtils is called', () => {
        const props = { ...commonProps, disabled: false };
        let instance;

        beforeAll(() => {
            instance = shallow(<DayInput { ...props } />).instance();
        });

        test('has short month name (without dot) in localized month string', () => {
            const januarySpain = instance.getLocaleUtils(true).getMonths('es')[0];
            expect(januarySpain).toEqual('ene');
        });

        test('has long month name (without dot) in localized month string', () => {
            const januaryFrance = instance.getLocaleUtils().getMonths('fr')[0];
            expect(januaryFrance).toEqual('janvier');
        });
    });

    describe('when Enter key is pressed', () => {
        const mockToday = new Date(Date.UTC(2000, 0, 1, 12));
        const eventForProps = ({ value }) => ({
            keyCode: KEYS.ENTER,
            target: { value },
            persist: jest.fn(),
            preventDefault: jest.fn(),
            stopPropagation: jest.fn()
        });

        describe('when input is empty', () => {
            const props = { ...commonProps, value: '' };
            const event = eventForProps(props);
            let component;

            beforeAll(() => {
                jest.spyOn(Date, 'now').mockImplementation(() => mockToday.getTime());
                component = mount(<DayInput { ...props } />);
            });

            afterAll(() => {
                jest.restoreAllMocks();
            });

            test('renders input field with today\'s date and calendar', () => {
                const input = component.find('input');
                input.prop('onKeyDown')(event);
                input.prop('onKeyUp')(event);
                component.update();
                expect(event.preventDefault).toHaveBeenCalledTimes(1);
                expect(component.find('.DayPickerInput-Overlay').exists()).toBeTruthy();
                expect(component.instance().dayPickerInput.state.value).toEqual('2000-01-01');
            });
        });

        describe('when date is specified', () => {
            const props = { ...commonProps, value: '2010-05-05' };
            const event = eventForProps(props);
            let component;

            beforeAll(() => {
                jest.spyOn(Date, 'now').mockImplementation(() => mockToday.getTime());
                component = mount(<DayInput { ...props } />);
                component.find('input').prop('onFocus')();
                component.update();
            });

            afterAll(() => {
                jest.restoreAllMocks();
            });

            test('renders input field with date and hides the calendar', () => {
                expect(component.find('.DayPickerInput-Overlay').exists()).toBeTruthy();

                const input = component.find('input');
                input.prop('onKeyDown')(event);
                input.prop('onKeyUp')(event);
                component.update();
                expect(event.preventDefault).toHaveBeenCalledTimes(1);
                expect(component.find('.DayPickerInput-Overlay').exists()).toBeFalsy();

                input.prop('onKeyDown')(event);
                input.prop('onKeyUp')(event);
                component.update();
                expect(event.preventDefault).toHaveBeenCalledTimes(2);
                expect(component.find('.DayPickerInput-Overlay').exists()).toBeTruthy();
                expect(component.instance().dayPickerInput.state.value).toEqual('2010-05-05');
            });
        });
    });
});
