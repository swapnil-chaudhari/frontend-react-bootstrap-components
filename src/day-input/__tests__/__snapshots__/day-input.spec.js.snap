// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DayInput> when rendered with required props renders <DayInput> with the right props 1`] = `
Object {
  "classNames": Object {
    "container": "DayPickerInput",
    "overlay": "DayPickerInput-Overlay",
    "overlayWrapper": "DayPickerInput-OverlayWrapper",
  },
  "clickUnselectsDay": false,
  "component": [Function],
  "dayPickerProps": Object {
    "canChangeMonth": true,
    "captionElement": [Function],
    "enableOutsideDaysClick": false,
    "fromMonth": undefined,
    "locale": "en",
    "localeUtils": Object {
      "formatDate": [Function],
      "formatDay": [Function],
      "formatMonthTitle": [Function],
      "formatWeekdayLong": [Function],
      "formatWeekdayShort": [Function],
      "getFirstDayOfWeek": [Function],
      "getMonths": [Function],
      "parseDate": [Function],
    },
    "modifiers": Object {
      "disabled": Object {
        "after": undefined,
        "before": undefined,
      },
    },
    "month": undefined,
    "navbarElement": [Function],
    "showOutsideDays": false,
    "toMonth": undefined,
    "todayButton": "todayLabel",
  },
  "format": "YYYY-MM-DD",
  "formatDate": [Function],
  "hideOnDayClick": true,
  "inputProps": Object {
    "className": "DateField-form-control form-control DayPickerInput-input",
    "disabled": false,
    "name": "my-day-input",
    "onBeforeKeyDown": [Function],
    "onKeyUp": [Function],
  },
  "keepFocus": true,
  "onDayChange": [Function],
  "overlayComponent": [Function],
  "parseDate": [Function],
  "placeholder": "YYYY-MM-DD",
  "showOverlay": false,
  "value": "",
}
`;
